docs/configurations.md

```diff
@@ -0,0 +1,77 @@
+BALENA_SUPERVISOR_DELTA_RETRY_COUNT | 30 | Number of retry attempts for downloading deltas. | v6.2.0
+BALENA_SUPERVISOR_DELTA_RETRY_INTERVAL | 10000 | Millisecond time before another delta download is attempted after failiure. | v6.2.0
+BALENA_SUPERVISOR_DELTA_VERSION | 3 | Specifies the delta version to use which has different APIs and download algorithms. | v7.9.0
+BALENA_SUPERVISOR_PERSISTENT_LOGGING | false | Persists journal logs to disk so they will be available after a reboot event. | v7.15.0
+BALENA_SUPERVISOR_INSTANT_UPDATE_TRIGGER | true | Enable / Disable instant triggering of updates when a new release is deployed. If set to false, the device will ignore the notification that is triggered when the device's target state has changed. In this case, the device will rely on polling to apply updates. Coupled with a large BALENA_SUPERVISOR_POLL_INTERVAL, this allows spreading out updates in large fleets to avoid overloading local networks when there is a large number of devices at one location. | v9.13.0
+BALENA_HOST_FIREWALL_MODE | off | Toggle firewall modes between on, off, and auto. | v11.9.1
+BALENA_HOST_DISCOVERABILITY | true | Enable / Disable Avahi to run which will allow the device to respond to requests such as network scans. | v11.9.2
+BALENA_HOST_SPLASH_IMAGE | /boot/splash/balena-logo-default.png | Allows changing splash screen on boot to user defined file from userspace. | v12.3.0
+---
+
+In addition to these values, there may be some device-type specific configuration variables that can be set.
+
+## Config.txt
+
+> Supported boards: Any Raspberry Pi and BalenaFin.
+
+This device type utilizes a `config.txt` [file](https://www.raspberrypi.org/documentation/configuration/config-txt/README.md) which we offer some values to be configured.
+
+Name | Default | Description | Valid from
+--- | --- | --- | ---
+BALENA_HOST_CONFIG_disable_splash | 1 | Enable / Disable the Raspberry Pi rainbow splash screen. | v1.0.0
+BALENA_HOST_CONFIG_dtparam | "i2c_arm=on","spi=on","audio=on" | Define DT parameters. | v1.0.0
+BALENA_HOST_CONFIG_gpu_mem | 16 | Define device GPU memory in megabytes. | v1.0.0
+
+You can find more information on updating `config.txt` through configuration variables in our [Advanced Boot Configuration Guide][boot-config-guide].
+
+## ConfigFS
+
+> Supported boards: UP Board
+
+This device type utilizes a `configFS` [file](https://www.raspberrypi.org/documentation/configuration/config-txt/README.md) interface that allows definition of arbitrary functions and configurations.
+
+Name | Default | Description | Valid from
+--- | --- | --- | ---
+BALENA_HOST_CONFIGFS_ssdt | undefined | Allows configuring a user defined SSDTs from userspace via the configfs interface. See [SSDT Overlays](https://www.kernel.org/doc/html/latest/admin-guide/acpi/ssdt-overlays.html#ssdt-overlays). | v10.9.2
+
+## Extra-uEnv
+
+> Supported boards: Intel NUC, Nvidia Jetson Nano and TX2.
```

Comment:
Only NVidia Jetson Nano and TX2 here, without Intel NUC

Reply:
I pulled these values from the device types the Supervisor will actually match the backend with and Intel NUC is one of them https://github.com/balena-os/balena-supervisor/blob/0f906fd9d8f4f7dc2bfd9481e882eb47f2ebbba5/src/config/backends/extra-uEnv.ts#L61-L69

Should we change this ?

Reply:
messaging you to discuss if we should change the deviceTypes. Resolving this though as the devices mentioned are what the code is going to match this.

Reply:
Hi @20k-ultra, I see, so this backend matches the NUC too. However, this won't work because the NUC is x86 which uses grub, so it doesn't support device-trees and implicitly no extlinux file. The configuration file which can specify extra cmdline arguments for the NUC is named grub_extraenv (see internal discussion https://www.flowdock.com/app/rulemotion/r-supervisor/threads/Vs_mpEKwt6K_kvgQ-wGw6ZH9A930) and locations for it in the boot partition are https://github.com/balena-os/balena-intel/blob/315071b5e0d98d7dd776a8b662ee8c0f73595e18/layers/meta-balena-genericx86/recipes-core/images/resin-image.inc#L18

So, on the NUC the extra_os_cmdline can't work as it is because the configuration file differs in name and path, and custom device-tree can't work because there's no device tree support on this platform. Therefore, the NUC should not be mentioned in the docs for these 2 configuration options.

Reply:
In PR https://github.com/balena-os/balena-supervisor/pull/1421 I noted that extra_uEnv backend is supported because of https://github.com/balena-os/balena-intel/pull/311. Was there a change or was the Intel Nuc never suppose to be there ? I've prempively created https://github.com/balena-os/balena-supervisor/pull/1659 to approve once you confirm that this should be removed.

Reply:
@20k-ultra just to clarify, does the supervisor use the 'grub_extraenv' file specifically for the Intel NUC somewhere in the sources, or does it use the same file extra_uEnv.txt for both jetson and NUC? It looks like it uses extra_uEnv.txt for the NUC too.

This PR https://github.com/balena-os/balena-intel/pull/311 mentions the grub environment file name "grub_extraenv"  and paths https://github.com/balena-os/balena-intel/pull/311/commits/5f8856bbef645b63cfa04593c5cad28f1586d467#diff-8ad673b0f5c6b06dca41dc93a32e63d4a66ff69ee5237718972a8687ae339084R20 , which aren't the same as for  ["extra_uEnv.txt"](https://github.com/balena-os/balena-supervisor/pull/1659/commits/e6eda0fca7b280aa5453972a26c36ab6895c227a#diff-7e32acc477451dde576c1cedba193ccc078e0f1068440be5b91571c059a3dc88L45).

I'm not aware of any change on the Intel NUC side, grub_extraenv should be the same as when it was added in [PR 311](https://github.com/balena-os/balena-intel/pull/311). So if we would want to have extra_os_cmdline="x,y,z" capability in SV for the intel NUC, https://github.com/balena-os/balena-supervisor/pull/1659/commits/e6eda0fca7b280aa5453972a26c36ab6895c227a#diff-7e32acc477451dde576c1cedba193ccc078e0f1068440be5b91571c059a3dc88L45 should mention these two paths specifically for the NUC:

```
<boot_partition>/grub/grub_extraenv \
<boot_partition>/EFI/BOOT/grub_extraenv \
```
 and leave untouched the extra_uEnv.txt file that's currently used for the jetson devices. 

extra_os_cmdline arguments would be the only option compatible with the NUC. odmdata and fdt are not supported on the x86 platform for the reasons I already mentioned.

Reply:
the extra_uenv backend does not use a different file for the intel nuc. I've removed it from this backend with https://github.com/balena-os/balena-supervisor/pull/1659 which is merged. Thanks for pointing this out. I guess I added it my some misunderstanding when I first wrote the feature.